linux/disk.go

```diff
@@ -384,18 +381,34 @@ func addPartitionSet(d disko.Disk, pSet disko.PartitionSet) error {
 		return fmt.Errorf("failed to stat %s: %s", d.Path, err)
 	}
 
+	// Call addpart if this is a block device.
 	if info.Mode()&os.ModeDevice != 0 {
-		// Call partx if this is a block device.
-		for _, n := range pnums {
-			if err := runCommand("partx", "--add", fmt.Sprintf("%d", n), d.Path); err != nil {
+		// Close the filehandle and release the lock, then call udevSettle
+		// so that we can call addpart if needed.
+		fp.Close()
+
+		if err := udevSettle(); err != nil {
+			return err
+		}
+
+		for _, p := range pSet {
+			ppath := getPartPathForKname(d.Name, p.Number)
+			if exists, _ := blockDeviceExists(ppath); exists {
```

Comment:
Do you want to check this error?

Reply:
maybe this is ugly, probably it is.
but blockDeviceExists will only ever return exists = true if there are no errors.

so if exists is true, we can ignroe the error.

if its false, then i figured we'd fall through and fail later.

Reply:
io'm open to suggestions on how to do it better.

Reply:
That violates "fail early", but it's really up to you.

Reply:
Why not just do the standard `if err != nil { return err }`?

Reply:
https://paste.ubuntu.com/p/9KKY8ywz79/
applied ^
thank you for your review.